x/mongo/driver/topology/pool.go

```diff
@@ -601,3 +527,392 @@ func (p *pool) clear(serviceID *primitive.ObjectID) {
 	}
 	p.generation.clear(serviceID)
 }
+
+func (p *pool) queueForIdleConn(w *wantConn) bool {
```

Comment:
nit: I find this function name a little confusing because it may or may not put `w` in the queue. Consider renaming to something like `getIdleConnOrQueue`. I have no strong feelings about this though, so feel free to leave it as-is.

Reply:
Seems reasonable, will change.